### PR TITLE
A provider run is audited as the provider it is for

### LINKED ISSUE #2843: A provider run's audit actor is hard-coded to connector:surfe
## What is wrong

`providerJobActor` in `backend/internal/compose/jobs_providerruns.go` binds every
provider run's principal to the literal `connector:surfe`, whichever provider the
run is actually for:

```go
ctx = principal.WithActor(ctx, principal.Principal{
    Type: principal.PrincipalSystem, ID: "connector:surfe",
})
```

Both the submit worker and the poll worker use it.

The claim rows themselves are correct — `people.WriteProviderClaims` derives
`captured_by` as `'connector:' || provider` from the run's own provider, so the
evidence on the record names the real vendor. Only the **audit actor** is wrong.

## Why it matters now

Until recently only one provider could exist: `provider_connection`,
`provider_run` and `person_provider_claim` each carried a
`CHECK (provider = 'surfe')`. Those constraints have been dropped so a second
data provider can be connected, and the person record now shows one section per
provider. With a second provider connected, its runs will write audit entries
attributing the purchase to Surfe while the claim rows name the real provider —
the audit log and the evidence disagree about who acted.

## Why it was deferred

The fix belongs in `modules/integrations`, not in the compose wiring: the submit
worker knows a run id but not its provider, and the poll worker drains many runs
at once, so the actor has to be resolved per run inside the store (which already
has `runProviderName`). That is a different module from the change that surfaced
it.

No second adapter is compiled today, so nothing misattributes yet.

## What it means if nobody fixes it

The first installation to connect a second provider gets an audit trail that
names the wrong actor for every purchase it makes. Auditability is one of the
guardrails the rulebook does not suspend, and an audit entry that names the
wrong actor is worse than a missing one because it reads as authoritative.

### PR DESCRIPTION
**Pull request by dungbanguyen**
Closes #2843.

## What was wrong

`providerJobActor` bound every provider run's principal to the literal `connector:surfe`, whichever provider the run was actually for. Both the submit worker and the poll worker used it.

The claim rows were already correct — `people.WriteProviderClaims` derives `captured_by` as `'connector:' || provider` from the run's own provider. Only the **audit actor** was wrong, so with a second provider connected the audit log and the evidence on the record would name different vendors for one purchase. An audit entry that names the wrong actor is worse than a missing one, because it reads as authoritative.

## Why the fix is where it is

The workers cannot know. The submit worker holds a run id; the poll sweep drains many runs at once. A principal bound out there can only ever name a vendor it guessed — and it guessed the one provider that could exist while `provider_connection`, `provider_run` and `person_provider_claim` each carried a `CHECK` pinning them to a single name. Those constraints have been dropped.

The store already resolves the provider before it does anything else with a run, at all three entry points (`ExecuteSubmit`, `pollOne`, `recoverClaims`), so it narrows the actor there — `actingForProvider`, in `integrations`, spelling the id with the same `connector:` prefix the claim rows carry, so a reader joining the two sees one name.

Compose keeps binding a principal, for the reason it was added: without an actor the hand-off refused with "no actor bound to context" and every polled run stayed `in_progress` forever, paid for and reaching nobody. It now names the **worker** rather than a vendor, so it cannot be mistaken for one if a path ever reaches a gated write before resolving a provider.

## What holds it

**`TestNoPrincipalNamesAConnectorInAStringLiteral`** — no principal anywhere in the tree may take a `connector:` id from a string literal. This needs a gate rather than just a fix: a literal is not a bug until there are two, nothing failed while one vendor was the only one, and the disagreement appears for the first time on the first installation to connect a second. The census reads the ID **field** rather than the file's strings (`connector:` also appears in provenance writers, prefix tests and SQL, none of which binds an actor), and resolves the principal package through `gatekit.ImportedAs` so an alias reads the same.

`connector:finance` is waived with its reason: it is the finance mirror's own name, not a vendor's — one ledger sweep per installation, with nothing to read a name from. `AssertAllMatched` keeps that honest.

**`TestARunAuditsTheProviderItIsFor`** — drives a run for a vendor that is deliberately **not** `surfe`, because a fixture carrying the one name cannot tell a derived answer from a constant that happens to match it. That required teaching the offline fake to stand in for another vendor (`Named`) and the runs fixture to seed its connection for it.

Mutation-checked: restoring the hard-coded actor at either layer fails the integration test naming the wrong actor, and writing a literal into `actingForProvider` fails the gate.

## Checks

`make check-backend`, `make craft-static` and the full `make -C backend test-integration` (48 packages, 0 skips) green locally.

## Summary by CodeRabbit

- **New Features**
  - Provider-run activity now uses the correct provider-specific connector identity for audit records.
  - Multi-provider integration runs can be configured and tracked independently.
  - Offline providers can be assigned custom provider names.

- **Bug Fixes**
  - Improved attribution for integration actions, polling, claim handoffs, submissions, and refusal events.
  - Removed provider-specific constraints that could prevent valid connector configurations.

- **Documentation**
  - Updated the changelog and gate inventory to reflect the new connector attribution behavior.

### CODE REVIEW

File: backend/gates/connectoractor_test.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+		}
+		for _, element := range lit.Elts {
+			field, isField := element.(*ast.KeyValueExpr)
+			if !isField {
+				continue
+			}
+			if key, isIdent := field.Key.(*ast.Ident); !isIdent || key.Name != "ID" {
+				continue
+			}
+			value, isBasic := field.Value.(*ast.BasicLit)
```
[the remark this answers is not in the document]
Comment (dungbanguyen) [reply backend/gates/connectoractor_test.go]: Agreed, and the wording gave the game away: a const is written down, not read from anywhere, so moving the literal one line further off is the same defect.

The scan walks a PACKAGE at a time now and folds its own string constants, because the const and its use are usually in different files. Mutation-checked with `const mutVendorActor = "connector:surfe"` used as an ID: the census names it.

File: backend/gates/connectoractor_test.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+		lit, isLit := n.(*ast.CompositeLit)
+		if !isLit || !namesAPrincipal(lit.Type, qualifier, dotImported) {
+			return true
+		}
+		for _, element := range lit.Elts {
+			field, isField := element.(*ast.KeyValueExpr)
+			if !isField {
+				continue
+			}
+			if key, isIdent := field.Key.(*ast.Ident); !isIdent || key.Name != "ID" {
```
[the remark this answers is not in the document]
Comment (dungbanguyen) [reply backend/gates/connectoractor_test.go]: Right, and the `read == 0` observation is the important half — the floor would have stayed green while a positional literal walked past.

A principal literal with unkeyed elements is now refused outright, once per literal, naming the file: this reader matches the ID field by name, so a positional one is a literal it cannot see, and answering "no id here" about it is the failure a census may not have. Mutation-checked with a full nine-field positional `principal.Principal` carrying `connector:surfe`.

File: backend/internal/modules/integrations/execute_integration_test.go [hunk 3]
```diff
@@ -514,3 +514,60 @@ func TestARefusalAuditsTheConnectionStatusItChangedFrom(t *testing.T) {
...
+// name different vendors for one purchase, and the entry that reads as
+// authoritative is the wrong one.
+//
+// The environment is deliberately NOT surfe, because a fixture carrying the one
+// name cannot tell a derived answer from a constant that happens to match.
+func TestARunAuditsTheProviderItIsFor(t *testing.T) {
+	// The credential is refused, because that is the submission outcome whose
+	// settlement writes an audit row: the connection's status changes, and the
+	// row says who observed it.
+	e := setupRuns(t, runsConfig{provider: "otherco", subjectLastName: "InvalidCredentials"})
```
[the remark this answers is not in the document]
Comment (dungbanguyen) [reply backend/internal/modules/integrations/execute_integration_test.go:535]: Fixed — the query and the expectation both come from `e.provider` now (`"connector:" + e.provider`), so changing the fixture updates the assertion instead of leaving it matching nothing behind the guard.

File: backend/internal/modules/integrations/runactor.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+// actingForProvider narrows the acting principal to the connector this run
+// belongs to, and stamps a correlation id if the caller has not.
+//
+// It REPLACES whatever the worker bound. The worker's principal exists so an
+// RBAC-gated write is never attempted with no actor at all; it cannot name a
+// vendor, and this is the first point in the run where one is known.
+func actingForProvider(ctx context.Context, name string) context.Context {
+	ctx = principal.WithActor(ctx, principal.Principal{
+		Type: principal.PrincipalSystem, ID: connectorActorPrefix + name,
```
[the remark this answers is not in the document]
Comment (dungbanguyen) [reply backend/internal/modules/integrations/runactor.go:46]: The inconsistency is real and I am not fixing it here, because the fix is not the one-line change it looks like.

`PrincipalSystem` is unbounded by design and `PrincipalConnector` is not. `auth.Require` returns nil unconditionally for a system principal (`rbac.go:68`); every other type falls through to `Permissions.Allows`, which a zero `Permissions` denies. `auth/rowscope.go` and `auth/inheritedscope.go` each take a system branch that skips the owner/audience predicate entirely. So switching the type means enumerating the fixed minimum grant set for the whole provider-run path — the ledger, the connection, the budget rows, the claim writes, the person reads — the way `financeSweepPrincipal` spells out its own.

Getting that set wrong has exactly the signature this area already produced once: `auth.Require` refuses, the job is discarded after its attempts, and the installation's enrichment silently stops with nothing on any screen to say why. That is the failure class this PR and #2842 are both about, and I am not going to introduce it while fixing it.

This PR changes WHICH connector a run is audited as, and does not touch the principal type — the id was wrong for a second vendor today, the type is the same shape it has always been. Filed as #3228 with the analysis and what a real fix needs, including a gate so the two spellings cannot both come back.

File: backend/gates/connectoractor_test.go [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+// should delete the entry rather than widen it.
+var oneOfItsKindForever = gatekit.Waive(map[string]string{
+	"connector:finance": "the finance mirror's own name, not a vendor's: one ledger sweep per installation, converting the source ledger into the base currency as it writes. There is nothing to read the name from, because there is never more than one",
+})
+
+// TestNoPrincipalNamesAConnectorInAStringLiteral is the census.
+func TestNoPrincipalNamesAConnectorInAStringLiteral(t *testing.T) {
+	t.Parallel()
+	read := 0
+	err := filepath.WalkDir("internal", func(walked string, d fs.DirEntry, err error) error {
```
[the remark this answers is not in the document]
Comment (dungbanguyen) [reply backend/gates/connectoractor_test.go]: This comment arrived with its header and its static-analysis block but no finding text, so there is nothing here to act on. If a claim was meant to accompany it, please restate it and I will.

Answering what the probe appears to have been checking: `filepath.WalkDir("internal", ...)` is the convention every census in this package uses, and it resolves because `TestMain` in `gates/gates_test.go:42` chdirs to `..` — the backend module root — before any test runs. `isIntegrationTagged` and `isPrincipalType` are existing helpers in this package; the latter is reused rather than redeclared, which is what made the first version of this file fail to build. There is no nested `internal` tree or symlink under `backend/gates`.

Resolving as nothing-to-address rather than as fixed.

File: backend/gates/connectoractor_test.go [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+				}
+				if key, isIdent := field.Key.(*ast.Ident); !isIdent || key.Name != "ID" {
+					continue
+				}
+				if text, ok := stringValue(field.Value); ok {
+					out = append(out, writtenActorID{file: path, id: text})
+					continue
+				}
+				if ident, isIdent := field.Value.(*ast.Ident); isIdent {
+					if text, declared := constants[ident.Name]; declared {
```
[the remark this answers is not in the document]
Comment (dungbanguyen) [reply backend/gates/connectoractor_test.go]: Both taken, and they are the same gap at different distances — so rather than add two more cases, the reader now indexes EVERY name a package binds to a string: package const or var, function-local assignment, const-block repetition, and aliases of any of those. Resolution runs to a fixpoint, so an alias settles once its target does.

Names are indexed per PACKAGE rather than per scope, which means two functions binding the same name share an entry. That can only produce a false POSITIVE — a literal reported at a site that did not use it — which is a loud failure a reader resolves in a minute, where the other direction is the one a census may not have. The header says so.

The folding goes through `gatekit.StringExpr` with `FoldStrict` rather than my own reader; `TestNobodyWritesASecondStringReader` caught the private one and it was right to, since a census reaching for its own answer to "what string is this" inherits whatever that answer cannot see.

Mutation-checked all three shapes: `vendor := "connector:surfe"` in a function, `const alias = vend`, and a bare `mutVendB` repeating the const above it. Each is named by the census.

File: backend/gates/connectoractor_test.go [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+			for _, s := range spec.Specs {
+				value, isValue := s.(*ast.ValueSpec)
+				if !isValue {
+					continue
+				}
+				for i, ident := range value.Names {
+					if i >= len(value.Values) {
+						continue
+					}
+					if text, ok := stringValue(value.Values[i]); ok {
```
[the remark this answers is not in the document]
Comment (dungbanguyen) [reply backend/gates/connectoractor_test.go]: Both taken, and they are the same gap at different distances — so rather than add two more cases, the reader now indexes EVERY name a package binds to a string: package const or var, function-local assignment, const-block repetition, and aliases of any of those. Resolution runs to a fixpoint, so an alias settles once its target does.

Names are indexed per PACKAGE rather than per scope, which means two functions binding the same name share an entry. That can only produce a false POSITIVE — a literal reported at a site that did not use it — which is a loud failure a reader resolves in a minute, where the other direction is the one a census may not have. The header says so.

The folding goes through `gatekit.StringExpr` with `FoldStrict` rather than my own reader; `TestNobodyWritesASecondStringReader` caught the private one and it was right to, since a census reaching for its own answer to "what string is this" inherits whatever that answer cannot see.

Mutation-checked all three shapes: `vendor := "connector:surfe"` in a function, `const alias = vend`, and a bare `mutVendB` repeating the const above it. Each is named by the census.

File: backend/gates/connectoractor_test.go [hunk 8]
```diff
@@ -0,0 +1,276 @@
...
+		}
+		files[path] = file
+		indexBindings(file, bound)
+	}
+	// A fixpoint, because an alias resolves only once its target has: the
+	// first pass folds the literals, the next folds the names pointing at
+	// them, and so on until nothing new resolves. The bound is what ends a
+	// cycle the compiler would refuse but this reader does not typecheck.
+	constants := map[string]string{}
+	for pass := 0; pass < 8; pass++ {
```
[the remark this answers is not in the document]
Comment (dungbanguyen) [reply backend/gates/connectoractor_test.go]: Right — `!progressed` already ends both a cycle and a chain, so the cap could only truncate a legitimate one and drop its tail out of the census without a sound. Gone. Mutation-checked with a fourteen-deep const chain ending in `connector:surfe`: the census names it.

File: backend/gates/connectoractor_test.go [hunk 9]
```diff
@@ -0,0 +1,276 @@
...
+			}
+			if text, ok := gatekit.StringExpr(expr, constants, gatekit.FoldStrict); ok {
+				constants[name] = text
+				progressed = true
+			}
+		}
+		if !progressed {
+			break
+		}
+	}
```
[the remark this answers is not in the document]
Comment (dungbanguyen) [reply backend/gates/connectoractor_test.go]: Right — `!progressed` already ends both a cycle and a chain, so the cap could only truncate a legitimate one and drop its tail out of the census without a sound. Gone. Mutation-checked with a fourteen-deep const chain ending in `connector:surfe`: the census names it.

File: backend/gates/connectoractor_test.go (hunk 8 above)
[the remark this answers is not in the document]
Comment (dungbanguyen) [reply backend/gates/connectoractor_test.go]: Right — `!progressed` already ends both a cycle and a chain, so the cap could only truncate a legitimate one and drop its tail out of the census without a sound. Gone. Mutation-checked with a fourteen-deep const chain ending in `connector:surfe`: the census names it.

File: backend/gates/connectoractor_test.go [hunk 10]
```diff
@@ -0,0 +1,276 @@
...
+				}
+				values := value.Values
+				if len(values) == 0 {
+					values = carried
+				} else {
+					carried = values
+				}
+				for i, ident := range value.Names {
+					if i < len(values) {
+						bound[ident.Name] = values[i]
```
[the remark this answers is not in the document]
Comment (dungbanguyen) [reply backend/gates/connectoractor_test.go]: Both of you are right, and the comment claiming this "can only produce a false POSITIVE" was itself the bug: a single entry per name means a later binding overwrites an earlier one, so `vendor := chosen` in an unrelated function masks `vendor := "connector:surfe"` used as an actor id and the literal disappears silently.

Every binding is kept now, and a connector value WINS the entry. That keeps the asymmetry the comment claimed and did not deliver: the census can report a connector name at a site that did not use it — loud, and a minute to resolve — but it can no longer lose one.

Full lexical scoping would be the exact answer, and it is a type-checked resolution rather than a syntax walk. This is the sound approximation in the direction that matters, and the header now says which direction that is. Mutation-checked with your exact shape: the masked literal is named.

File: backend/gates/connectoractor_test.go [hunk 11]
```diff
@@ -0,0 +1,276 @@
...
+					}
+				}
+			}
+		case *ast.AssignStmt:
+			for i, lhs := range node.Lhs {
+				ident, isIdent := lhs.(*ast.Ident)
+				if isIdent && i < len(node.Rhs) {
```
[the remark this answers is not in the document]
Comment (dungbanguyen) [reply backend/gates/connectoractor_test.go:214]: Both of you are right, and the comment claiming this "can only produce a false POSITIVE" was itself the bug: a single entry per name means a later binding overwrites an earlier one, so `vendor := chosen` in an unrelated function masks `vendor := "connector:surfe"` used as an actor id and the literal disappears silently.

Every binding is kept now, and a connector value WINS the entry. That keeps the asymmetry the comment claimed and did not deliver: the census can report a connector name at a site that did not use it — loud, and a minute to resolve — but it can no longer lose one.

Full lexical scoping would be the exact answer, and it is a type-checked resolution rather than a syntax walk. This is the sound approximation in the direction that matters, and the header now says which direction that is. Mutation-checked with your exact shape: the masked literal is named.
